services/report/parser/__init__.py

```diff
@@ -3,7 +3,9 @@
 from services.report.parser.version_one import VersionOneReportParser
 
 
-def get_proper_parser(upload: Upload):
+def get_proper_parser(upload: Upload, parallel_idx=None):
+    if parallel_idx is not None:
+        return LegacyReportParser()
```

Comment:
during the old behaviour, processor task will [rewrite the raw uploaded coverage report](https://github.com/codecov/worker/blob/a01c215e950db16f121df51ce79a86420edf4c6c/tasks/upload_processor.py#L259) to a human readable format. Hence, when the parallel flow starts after serial flow has completely finished, it will have to parse the human readable format instead of the raw uploaded report.

Looking back, maybe it would be better for the experiment to use the raw uploaded report format as it is closer to what we would expect when we fully rollout parallel upload processing